addons/account/models/account_move.py

```diff
@@ -3037,6 +3030,26 @@ def _get_update_invoice_from_attachment_decoders(self, invoice):
         """
         return []
 
+    @api.model
+    def _get_company_terms(self, company):
+        if not company.terms_type == 'html':
+            return company.invoice_terms if not is_html_empty(company.invoice_terms) else ''
+        baseurl = self.env.company.get_base_url() + '/terms'
+        return _('Terms & Conditions: %s', baseurl)
+
+    @api.depends('move_type', 'partner_id')
+    def _compute_narration(self):
+        use_invoice_terms = self.env['ir.config_parameter'].sudo().get_param('account.use_invoice_terms')
+        # Avoid using the default TC for vender bills
+        is_sale = self.env.context.get('default_move_type') in self.env['account.move'].get_sale_types()
+        force_empty_tc = not use_invoice_terms or not is_sale
+        for move in self.filtered(lambda am: not am.narration):
+            if force_empty_tc or not move.is_sale_document(include_receipts=True):
+                move.narration = ''
+            else:
+                narration_context = self.env['account.move'].with_context(lang=move.partner_id.lang or self.env.lang)
+                move.narration = narration_context._get_company_terms(move.company_id) or narration_context._get_company_terms(self.env.company)
```

Comment:
`depends` on `company_id` and you shouldn't need the `or`

Reply:
Thanks for the suggestions. As it is useless, I removed the _get_company_terms method to do the logic in the compute.